pennylane/gradients/finite_difference.py

```diff
@@ -24,7 +24,11 @@
 
 import pennylane as qml
 
-from .gradient_transform import gradient_transform
+from .gradient_transform import (
+    gradient_transform,
+    _grad_method_validation,
+    _choose_params_with_methods,

```

Comment:
Should these two have no leading underscore, given they are used across the module now?

Reply:
definitely :) It might also be worth giving `_choose_params_with_methods` a better name.

Reading the function name should give _some_ indication to what the function does, but this one is pretty obscure

Reply:
What do you think about `choose_grad_methods`? Because this method basically chooses grad methods from a set of grad methods, based on the provided `argnum`

Reply:
Reads well to me!